src/components/Popup.tsx

```diff
@@ -0,0 +1,200 @@
+    hidden : boolean}> = (props) => {
+    const history = useNavigate();
+    const [nameDirty, setNameDirty] = useState(-1)
+    const [numberDirty, setNumberDirty] = useState(-1)
+    const [addressDirty, setAddressDirty] = useState(-1)
+    const [emailDirty, setEmailDirty] = useState(-1)
+    const [cardDirty, setCardDirty] = useState(-1)
+    const [validDirty, setValidDirty] = useState(-1)
+    const [codeDirty, setCodeDirty] = useState(-1);
+    const [cvv, setCvv] = useState(300);
+    const [hidden, setHidden] = useState(false);
+    const [hidepop, setHidepop] = useState(false);
+
+
+    const isAllValid = () : boolean =>{
+        let isValid: number = 1;
+        if(!!nameDirty){
+            setNameDirty(1);
+            isValid = 0
+        }
+        if(!!numberDirty){
+            setNumberDirty(1);
+            isValid = 0
+        }
+        if(!!addressDirty){
+            setAddressDirty(1);
+            isValid = 0
+        }
+        if(!!emailDirty){
+            setEmailDirty(1);
+            isValid = 0
+        }
+        if(!!cardDirty){
+            setCardDirty(1);
+            isValid = 0
+        }
+        if(!!validDirty){
+            setValidDirty(1);
+            isValid = 0
+        }
+        if(!!codeDirty){
+            setCodeDirty(1);
+            isValid = 0
+        }
+        return !!isValid
+    }
+
+    function confrimForm(){
+        if(isAllValid()){
+            setHidepop(true)
+            localStorage.setItem("cart",'[]')
+            setTimeout(() => {
+                history("/")
+            }, 4000);
+            
+        }
+    }
+
+    let lastVal: string = ""
+    const setSlash = (e: {target: HTMLInputElement}):void => {
+        if(lastVal.length < e.target.value.length && e.target.value.length === 2) {
+            e.target.value += "/"   
+        }
+        lastVal = e.target.value 
+    }
+    const blurHandler = (e : string, value: string): void => {
+        switch (e) {
+            case 'name':
+              if(value.split(" ").length > 1){
+                setNameDirty((value.split(" ")[0].length > 3 && value.split(" ")[1].length > 3)?0:1)
+              }else{
+                setNameDirty(1)
+              }
+                break;
+            case 'number':
+                if(value[0] === "+" && value.length > 9 && parseInt(value.replaceAll(" ","")) + '' === value.replace("+","").replaceAll(" ","")){
+                    setNumberDirty(0)
+                }else{
+                    setNumberDirty(1)
+                }
+                
+                break;
+            case 'address':
+                if(value.split(" ").length > 2){
+                    let i = value.split(" ");
+                    setAddressDirty((i[0].length > 4 && i[1].length > 4 && i[2].length > 4)?0:1)   
+                }else{
+                    setAddressDirty(1)
+                }
+                break;
+            case 'email':
+                var re = /\S+@\S+\.\S+/;
```

Comment:
This is a bad naming. First of all avoid `var`. Secondly you can create constants.ts file and put it there.